server/storage/mvcc/kvstore.go

```diff
@@ -388,19 +388,30 @@ func (s *store) restore() error {
 		}
 	}
 
+	// NOTE: Last compaction could delete all the related revisions but it
+	// didn't have chance to update it finished state. When revisions in
+	// [N, `scheduledCompact`] are tombstone, the `currentRev` will be less
+	// than scheduledCompact. The next following `compactLockfree` call
+	// will return ErrFutureRev. Since we force-commit scheduledCompact
+	// rev, it's trusted value. We should update `currentRev` to align with
+	// `scheduledCompact` just in case that the revision is decreased.
```

Comment:
You can also link to issue for reference if you want.
```suggestion	
	// If the latest revision is a tombstone revision and etcd just compacted
	// it, but crashed right before persisting the FinishedCompactRevision,
	// then it would lead to revision decreasing in bbolt db file. In such
	// a scenario, we should adjust the current revision using the scheduled
	// compact revision on bootstrap when etcd gets started again.	
```

Reply:
Thanks! Updated.